code/modules/clothing/glasses/_glasses.dm

```diff
@@ -213,6 +213,22 @@
 	. = ..()
 	REMOVE_TRAIT(user, TRAIT_BOOZE_SLIDER, CLOTHING_TRAIT)
 
+/obj/item/clothing/glasses/sunglasses/advanced/reagent_2
+	name = "science sunglasses"
+	desc = "A pair of sunglasses outfitted with apparatus to scan reagents, as well as providing an innate understanding of liquid viscosity while in motion. Has enhanced shielding which blocks flashes."
+	icon_state = "sunhudscience"
+	item_state = "sunhudscience"
+	scan_reagents = TRUE
+

```

Comment:
I'm not that good at coding but doesn't this need these so they can do the shit the normal sci goggles can do and be acid-proof like them?

	actions_types = list(/datum/action/item_action/toggle_research_scanner)
	resistance_flags = ACID_PROOF
	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 100)

Reply:
Yeah they do actually